runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,391 @@
+
+    /**
+     * Constructor for LogReplicationLogicalGroupClient
+     *
+     * @param runtime Corfu Runtime
+     * @param clientName String representation of the client name
+     * @throws InvocationTargetException InvocationTargetException
+     * @throws NoSuchMethodException NoSuchMethodException
+     * @throws IllegalAccessException IllegalAccessException
+     */
+    public LogReplicationLogicalGroupClient(CorfuRuntime runtime, String clientName) throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        Preconditions.checkArgument(isValid(clientName), String.format("%s", "clientName is null or empty."));
+
+        this.corfuStore = new CorfuStore(runtime);
+        this.clientName = clientName;
+        this.clientKey = ClientRegistrationId.newBuilder()
+                .setClientName(clientName)
+                .build();
+        Instant time = Instant.now();
+        this.clientInfo = ClientRegistrationInfo.newBuilder()
+                .setClientName(clientName)
+                .setModel(model)
+                .setRegistrationTime(Timestamp.newBuilder().setSeconds(time.getEpochSecond()).setNanos(time.getNano()).build())
+                .build();
+
+        try {
+            this.replicationRegistrationTable = corfuStore.getTable(CORFU_SYSTEM_NAMESPACE, LR_REGISTRATION_TABLE_NAME);
+        } catch (NoSuchElementException nse) {
+            this.replicationRegistrationTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_REGISTRATION_TABLE_NAME, ClientRegistrationId.class,
+                    ClientRegistrationInfo.class, null, TableOptions.fromProtoSchema(ClientRegistrationInfo.class));
+        }
+
+        try {
+            this.sourceMetadataTable = corfuStore.getTable(CORFU_SYSTEM_NAMESPACE, LR_MODEL_METADATA_TABLE_NAME);
+        } catch (NoSuchElementException nse) {
+            this.sourceMetadataTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_MODEL_METADATA_TABLE_NAME, ClientDestinationInfoKey.class,
+                    DestinationInfoVal.class, null, TableOptions.fromProtoSchema(DestinationInfoVal.class));
+        }
+
+        register();
+    }
+
+    /**
+     * Adds client and model to LogReplicationRegistrationTable.
+     */
+    private void register() {
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientRegistrationInfo clientRegistrationInfo = txn.getRecord(replicationRegistrationTable, clientKey).getPayload();
+
+                    if (clientRegistrationInfo != null) {
+                        log.warn("Client already registered.\n--- ClientRegistrationId ---\n{}--- ClientRegistrationInfo ---\n{}", clientKey, clientRegistrationInfo);
+                    } else {
+                        txn.putRecord(replicationRegistrationTable, clientKey, clientInfo, null);
+                        txn.commit();
+                    }
```

Comment:
you seem to be missing a `commit()` if this condition is true. Please consider adding the commit() outside the `if` block?

Reply:
+1